testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -13063,3 +13065,425 @@ fn tenure_extend_cost_threshold() {
+    // make sure the tenure was won by RL2
+    assert!(tip.sortition, "No sortition was won");
+    assert_eq!(
+        tip.miner_pk_hash.unwrap(),
+        mining_pkh_2,
+        "RL2 did not win the sortition"
+    );
+
+    info!("------------------------- RL2 Proposes Block N' -------------------------");
+
+    wait_for(30, || {
+        Ok(proposed_blocks_rl2.load(Ordering::SeqCst) > proposals_before)
+    })
+    .expect("Timed out waiting for the block proposal from RL2");
+
+    info!("------------------------- Block N is Announced -------------------------");
+
+    TEST_BROADCAST_PROPOSAL_STALL.set(true);
+    TEST_P2P_BROADCAST_STALL.set(false);
+    let proposals_before = proposed_blocks_rl2.load(Ordering::SeqCst);
+
+    // Wait for RL2's tip to advance to the last block
+    wait_for(30, || {
+        let Some(chain_info_2) = get_chain_info_opt(&conf_node_2) else {
+            return Ok(false);
+        };
+        Ok(chain_info_2.stacks_tip_height == block_n.stacks_height)
+    })
+    .expect("Timed out waiting for RL2 to advance to block N");
+
+    info!("------------------------- RL2 Proposes Block N+1 -------------------------");
+    // Miner 2 should be interrupted from waiting for N' to be accepted when it sees N
+
+    info!("Stop signers from ignoring proposals");
+    TEST_IGNORE_ALL_BLOCK_PROPOSALS.set(Vec::new());
+    TEST_BROADCAST_PROPOSAL_STALL.set(false);
+
+    wait_for(30, || {
+        Ok(proposed_blocks_rl2.load(Ordering::SeqCst) > proposals_before)
+    })
+    .expect("Timed out waiting for the new block proposal from RL2");
+
+    info!("------------------------- Signers Accept Block N+1 -------------------------");
+
+    wait_for(30, || {
+        let Some(chain_info_2) = get_chain_info_opt(&conf_node_2) else {
+            return Ok(false);
+        };
+        Ok(chain_info_2.stacks_tip_height == block_n.stacks_height + 1)
+    })
+    .expect("Timed out waiting for RL2 to advance to block N+1");
+
+    info!("------------------------- Next Tenure Builds on N+1 -------------------------");
+
+    let rl1_commits_before = commits_submitted_rl1.load(Ordering::SeqCst);
+    let rl2_commits_before = commits_submitted_rl2.load(Ordering::SeqCst);
+
+    skip_commit_op_rl1.set(false);
+    skip_commit_op_rl2.set(false);
+
+    // Wait for both miners to submit block commits
+    wait_for(30, || {
+        Ok(
+            commits_submitted_rl1.load(Ordering::SeqCst) > rl1_commits_before
+                && commits_submitted_rl2.load(Ordering::SeqCst) > rl2_commits_before,
+        )
+    })
+    .expect("Timed out waiting for miners to submit block commits");
+
+    next_block_and_process_new_stacks_block(
+        &mut signer_test.running_nodes.btc_regtest_controller,
+        30,
+        &signer_test.running_nodes.coord_channel,
+    )
+    .expect("Timed out waiting for the next block to be mined");
+
+    wait_for_chains().expect("Timed out waiting for Rl1 and Rl2 chains to advance");
+
```

Comment:
Should you add a check for the `N+2` block to be built on top of `N+1`?

Reply:
Added in 5308a99778e7b0155f74691abb9eb0abf63b8b93.